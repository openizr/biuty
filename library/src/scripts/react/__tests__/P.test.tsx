/**
 * Copyright (c) Matthieu Jabbour. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import React from 'react';
import UIP from 'scripts/react/P';
import { act } from 'react-dom/test-utils';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');

describe('react/UIP', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    ((container as unknown) as null) = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<UIP label="Test" modifiers="large" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with id', () => {
    act(() => {
      render(<UIP label="Test" id="test" />, container);
    });
    expect(container).toMatchSnapshot();
  });

  test('renders correctly - with itemProp', () => {
    act(() => {
      render(<UIP label="Test" itemProp="description" />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
