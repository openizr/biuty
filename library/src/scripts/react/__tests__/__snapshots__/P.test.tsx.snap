// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/UIP renders correctly - basic 1`] = `
<div>
  <p
    class="ui-p ui-p--large"
  >
    Test
  </p>
</div>
`;

exports[`react/UIP renders correctly - with id 1`] = `
<div>
  <p
    class="ui-p"
    id="test"
  >
    Test
  </p>
</div>
`;

exports[`react/UIP renders correctly - with itemProp 1`] = `
<div>
  <p
    class="ui-p"
    itemprop="description"
  >
    Test
  </p>
</div>
`;
